src/main/webapp/script.js

```diff
@@ -53,6 +60,117 @@ async function fetchQuestions(page) {
   });
 }
 
+/**
+ * Fetches a single question and its answers from server, 
+ * wraps each in an <li> element, and adds them to the DOM.
+ */
+async function fetchAnswers() {
+  const question_id = (new URL(document.location)).searchParams.get("id");
+  const response = await fetch('/fetch-answers?id=' + question_id);
+  const answersObject = await response.json();
+  const answersContainer = document.getElementById('answers');
+  Object.values(answersObject).forEach(answer => {
+    answersContainer.appendChild(createAnswerElement(answer));
+    const commentsContainer = document.createElement('ul');
+    commentsContainer.setAttribute('class', 'list-group list-group-flush ml-5');
+    answer.commentList.forEach(comment => {
+      if (comment.body) {
+        commentsContainer.appendChild(createCommentElement(comment));
+      }
+    });
+    answersContainer.appendChild(commentsContainer);
+    answersContainer.appendChild(document.createElement('br'));
+  });
+}
+
+/**
+ * Displays navbar authentication buttons according to login status.
+ */
+function fetchAuthentication() {
+  fetch('/authentication').then(response => response.json()).then(user => {
+    if (user.isUserLoggedIn) {
+      // If user is logged in, show logout button in navbar.
+      if (!user.isUserRegistered) {
+        // If logged in user is not registered, redirect to signup page.
+        window.location.replace(user.authenticationUrl);
+      }
+
+      // Delete signup button.
+      const signupButtonNavbar = document.getElementById('signup');
+      signupButtonNavbar.innerHTML = '';
+
+      // Add logout button to navbar.
+      addAuthenticationButton(
+          user.authenticationUrl, 'btn-outline-success', 'Log Out', 'login');
+
+
+      const questionSubmission = document.getElementById('post-question');
+      if (questionSubmission) {
+        questionSubmission.style.display = "block";
+      }
```

Comment:
Don't think this belongs in this function. Maybe you could create another one?

Reply:
but that would mean calling the same servlet twice, wouldn't that be a bit inefficient?

Reply:
But why do you need to fetch the servlet? I don't see you using the JSON it responds with...

Reply:
I need to know whether the user is logged in or not to show the form

Reply:
Ooh, got it!